### ISSUE TITLE
stubtest false positive when `__class_getitem__` is implemented in class body and corresponding body in stub is generic

### ISSUE THREAD

**Issue by jonathandung** [python/mypy#21253]
**To Reproduce**

In `_mypy/_1.py`:

```python
class spam:
    def __class_getitem__(cls, _, /): return cls
```

In `_mypy/_1.pyi`:

```python
class spam[T]: ...
```

**Expected Behavior**

Stubtest should not report any errors. The class is implemented to be generic since a `__class_getitem__` method is provided, and type-checked to be generic from the type parameter syntax.

**Actual Behavior**

```text
error: _mypy._1.spam.__class_getitem__ is not present in stub
Stub: in file /_mypy/_1.pyi
MISSING
Runtime: in file /_mypy/_1.py:2
def (_, /)
```

**Your Environment**

- Mypy version used: 1.20.0
- Command used: python -m mypy.stubtest _mypy
- Python version used: 3.14

**Comment by GalSakuri** [comment]
opened a PR for this: #21258

**Comment by jonathandung** [comment]
Thanks
